tokio-trace-serde/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+use std::fmt;
+
+use serde::{
+    ser::{SerializeMap, SerializeSeq, SerializeStruct, Serializer},
+    Serialize,
+};
+
+use tokio_trace_core::{
+    event::Event,
+    field::{Field, FieldSet, Visit},
+    metadata::{Level, Metadata},
+    span::{Attributes, Id, Record},
+};
+
+#[derive(Debug)]
+pub struct SerializeField(Field);
+
+impl Serialize for SerializeField {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.0.name())
+    }
+}
+
+#[derive(Debug)]
+pub struct SerializeFieldSet<'a>(&'a FieldSet);
+
+impl<'a> Serialize for SerializeFieldSet<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
+        for element in self.0 {
+            seq.serialize_element(element.name())?;
+        }
+        seq.end()
+    }
+}
+
+#[derive(Debug)]
+pub struct SerializeLevel<'a>(&'a Level);
+
+impl<'a> Serialize for SerializeLevel<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if self.0 == &Level::ERROR {
+            serializer.serialize_str("ERROR")
+        } else if self.0 == &Level::WARN {
+            serializer.serialize_str("WARN")
+        } else if self.0 == &Level::INFO {
+            serializer.serialize_str("INFO")
+        } else if self.0 == &Level::DEBUG {
+            serializer.serialize_str("DEBUG")
+        } else if self.0 == &Level::TRACE {
+            serializer.serialize_str("TRACE")
+        } else {
+            unreachable!()
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SerializeId<'a>(&'a Id);
+
+impl<'a> Serialize for SerializeId<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Id", 1)?;
```

Comment:
I think we should use [`serialize_tuple_struct`](https://docs.serde.rs/serde/ser/trait.Serializer.html#tymethod.serialize_tuple_struct) here instead:
```suggestion
        let mut state = serializer.serialize_tuple_struct("Id", 1)?;
```

This would mean we don't have to give the field on the `Id` a name, and would more accurately reflect the way the type is actually implemented